File: lib/to_mdx/parsers/open_document/office/presentation.rb
# frozen_string_literal: true

require_relative "../base"
require_relative "./automatic_styles"

module ToMdx
  module Parsers
    module Office
      class Presentation < Base
        def to_mdx
          <<~MDX
            <Slides>
            #{contentful_children.map(&:to_mdx).join("\n")}
            </Slides>
          MDX
        end

        def to_s
          contentful_children.map(&:to_s).join
        end

        register_for "office:presentation"
      end
    end
  end
end

File: src/parsers/open-document/office/presentation.ts
import { BaseElement } from "../../base_element.ts";

export class Presentation extends BaseElement {
  toMdx(): string {
    const slidesContent = this.contentfulChildren()
      .map((child) => child.toMdx())
      .join("\n");

    return `<Slides>
${slidesContent}
</Slides>`;
  }

  toString(): string {
    return this.contentfulChildren()
      .map((child) => child.toString())
      .join("");
  }

  static {
    BaseElement.registerFor.call(this, "office:presentation");
  }
}
